mesonbuild/backend/backends.py

```diff
@@ -157,7 +157,7 @@ def get_target_dir(self, target):
 
     def get_target_source_dir(self, target):
         dirname = os.path.join(self.build_to_src, self.get_target_dir(target))
-        return dirname
+        return os.path.relpath(dirname)
```

Comment:
This will return the path relative to the current directory, which might work now but will break quietly for users.

If the problem is `self.get_target_dir()` returning an empty string, it's probably best to add an `if` condition for it.

Reply:
@nirbheek thanks, updated